datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StorageValidationService.java

```diff
@@ -112,6 +114,12 @@ public ValidationResult validateBackupStorage(SdxCluster sdxCluster) {
         }
         ValidationResult.ValidationResultBuilder validationResultBuilder = new ValidationResult.ValidationResultBuilder();
         try {
+            // When the backup location is provided, DetailedEnvironmentResponse is updated to have the validation on the location provided.
+            if (!Strings.isNullOrEmpty(backupLocation)) {
```

Comment:
Could you instead pass the location to the validator instead of modifying the response?

else were we might assume that the whole response is authoritative.